Woopra documentation (#889)
rudderlabs/rudderstack-docs: docs/destinations/streaming-destinations/woopra.mdx

---
title: "Woopra"
description: Step-by-step guide on sending your event data from RudderStack to Woopra.
---

[Woopra](https://www.woopra.com/) is a customer journey and product analytics tool. It lets you track your customers' activities across various product, marketing, and sales touchpoints. You can also leverage built-in triggers to take real-time actions based on user behavior.

RudderStack supports Woopra as a destination where you can send your event data seamlessly.

## Getting started

Before configuring Woopra as a destination in RudderStack, verify if the source platform is supported by Woopra by referring to the table below:

| **Connection Mode** | **Web**       | **Mobile**    | **Server**    |
| :------------------ | :------------ | :------------ | :------------ |

> | **Device mode**     |  **Supported**         | -             | -             |

Author: @anantjain45823 @shrouti1507 info related to the web device mode is not visible in the dashboard settings image, as discussed with Anant. We will update the image later.

</div>

Once you have confirmed that the source platform supports sending events to Woopra, follow these steps:

1. From your [RudderStack dashboard](https://app.rudderstack.com/), add the source. Then, from the list of destinations, select **Woopra**.
2. Assign a name to the destination and click **Continue**.

## Connection settings

To successfully configure Woopra as a destination, you need to configure the following settings:

<img src="../../assets/event-stream-destinations/woopra-connection-settings-1.png" alt="Woopra connection settings"/>
<img src="../../assets/event-stream-destinations/woopra-connection-settings-2.png" alt="Woopra connection settings"/>

- **Project Name**: Enter your Woopra project name. This is the same as the domain you entered while setting up the project.

<div class="infoBlock">
For more information on obtaining your Woopra project name, refer to the <Link to="#faq">FAQ</Link> section below.
</div>

<div class="infoBlock">
RudderStack uses this name as a fallback value if you do not provide it in your event's <code class="inline-code">integrations</code> object.

> </div>

Author: @anantjain45823 please verify if this note is correct. 

Still a bit confused as to whether RudderStack uses this setting as a fallback value if the user doesn't specify any event name in the payload? @shrouti1507

Reviewer: If the user is sending the project name of woopra via integrations object like below:
```
"integrations": {
        "woopra": {
          "projectName": "abc.com"
        }
      }
```
That will come under higher priority than whatever they have set over the webapp

Author: Thanks @shrouti1507 - this is applicable when sending events only via cloud mode right? or does web device mode support this feature too?

Reviewer 2: Only for cloud mode

| `traits.email` / `context.traits.email` / `properties.email` | `cv_email` | String | Required, if `cv_id` or `cookie` is absent. |
| `context.externalId.woodpraId` / `anonymousId` | `cookie` | String | Required, if `cv_email` or `cv_id` is absent. |
| `context.ip` | `cv_ip` | IP address | Optional |
| `traits.company` | `cv_company` | String | Optional |
| `traits.name` / `context.traits.name` | `cv_name` | String | Optional |
| `traits.{x}` / `context.traits.{x}` | - | Optional, sent to Woopra as custom fields. |
| `context.device.type` | `device` | String | Optional |
| `context.app.name` + `context.app.version` | `app` | String | Optional |
| `ua.browser.name` + `ua.browser.version` | `browser` | String | Optional |
| `context.os.name` | `os` | String | Optional |

RudderStack uses the below fields to identify a user, in the same precedence:

1. `userId` / `traits.userId` / `traits.id` / `context.traits.userId` / `context.traits.id` (mapped to `cv_id`)
2. `traits.email` / `context.traits.email` / `properties.email` (mapped to `cv_email`)
3. `context.externalId.woodpraId` / `anonymousId` (mapped to `cookie`)

> 

Reviewer 2: Please add a note here that for user provided cookie we have to map that cookie to either an email or cv_id . i.e. sending just the cookie for new visitor for any call won't work.

Author: @anantjain45823 could you please share an example of this?

</div>

RudderStack uses the below fields to identify a user in the same order of precedence:

| Precedence order | RudderStack property | Woopra property |
| :----| :---- | :----- |
| 1 | `userId` / `traits.userId` / `traits.id` / `context.traits.userId` / `context.traits.id` |  `cv_id` |
| 2 | `traits.email` / `context.traits.email` / `properties.email` | `cv_email` | 
| 3 | `context.externalId.woodpraId` / `anonymousId` | `cookie` |

<div class="infoBlock">
For a detailed list of the supported property mappings, refer to the <Link to="#standard-property-mappings">Standard property mappings</Link> section below.
</div>

Any user-provided cookie must be mapped to either `email` or `cv_id` when creating a new user or using the cookie for the first time. Otherwise, Woopra accepts the data but will reflect it **only when** that cookie is mapped to any of these identifiers.

For example, Woopra accepts the following event but does not reflect the data in its dashboard:

```javascript
rudderanalytics.identify({
  name: "Alex Keener",
  country: "USA",

>   anonymousId: "hiu31e23r2ffwef",

Author: @anantjain45823 to confirm this inclusion.

| :-----| :------| :-----|
| `properties.{x}` | `ce_{x}` | - |
| `name` | `event` | String (in lower case) |

## Standard property mappings

The following table lists the supported property mappings between RudderStack and Woopra, **common to all the above events**:

| RudderStack property | Woopra property | Data type | Presence |
| :-----| :------| :-----| :------|
| `integration.woopra.propertyName` | `property` | Domain | Optional |
| `userId` / `traits.userId` / `traits.id` / `context.traits.userId` / `context.traits.id` | `cv_id` | String | Required, if `cv_email` or `cookie` is absent. |
| `traits.email` / `context.traits.email` / `properties.email` | `cv_email` | String | Required, if `cv_id` or `cookie` is absent. |
| `context.externalId.woodpraId` / `anonymousId` | `cookie` | String | Required, if `cv_email` or `cv_id` is absent. |
| `context.ip` | `cv_ip` | IP address | Optional |
| `traits.company` | `cv_company` | String | Optional |
| `traits.name` / `context.traits.name` | `cv_name` | String | Optional |
| `traits.{x}` / `context.traits.{x}` | - | Optional, sent to Woopra as custom fields. |
| `context.device.type` | `device` | String | Optional |

> | `app` + `context.app.version` | `app` | String | Optional |

Author: ```suggestion
| `context.app.name` | `app` | String | Optional |
```

| **Connection Mode** | **Web**       | **Mobile**    | **Server**    |
| :------------------ | :------------ | :------------ | :------------ |
| **Device mode**     |  **Supported**         | -             | -             |
| **Cloud mode**      |  **Supported** | **Supported** |  **Supported** |

<div class="infoBlock">
To know more about the difference between cloud mode and device mode in RudderStack, refer to the <Link to="/destinations/rudderstack-connection-modes/">RudderStack Connection Modes</Link> guide.
</div>

Once you have confirmed that the source platform supports sending events to Woopra, follow these steps:

1. From your [RudderStack dashboard](https://app.rudderstack.com/), add the source. Then, from the list of destinations, select **Woopra**.
2. Assign a name to the destination and click **Continue**.

## Connection settings

To successfully configure Woopra as a destination, you need to configure the following settings:

<img src="../../assets/event-stream-destinations/woopra-connection-settings-1.png" alt="Woopra connection settings"/>

> - **Project Name**: Enter your Woopra project name. This is the domain you entered while setting up your project.

Reviewer 3: ```suggestion
- **Project Name**: Enter your Woopra project name. This is the domain name you entered while setting up your project.
```

<div class="infoBlock">
For more information on getting your Woopra project name, refer to the <Link to="#faq">FAQ</Link> section below.
</div>

<div class="infoBlock">

> RudderStack uses this project name as a fallback value if it not specified in the event's <code class="inline-code">integrations</code> object. For more information, refer to the <Link to="#identify">Identify</Link> section below.

Reviewer 3: ```suggestion
RudderStack uses this project name as a fallback value if it is not specified in the event's <code class="inline-code">integrations</code> object. For more information, refer to the <Link to="#identify">Identify</Link> section below.
```

    }
  }
});
```

You can specify your Woopra project name within the `integrations` object of the `identify` event, as seen in the above snippet. Alternatively, you can specify the project name via the **Project Name** dashboard setting while configuring the destination.

<div class="infoBlock">
If you provide the Woopra project name in both the dashboard settings as well as the <code class="inline-code">integrations</code> object, RudderStack gives precedence to the name specified in the <code class="inline-code">integrations</code> object.
</div>

RudderStack uses the below fields to identify a user in the same order of precedence:

| Precedence order | RudderStack property | Woopra property |
| :----| :---- | :----- |
| 1 | `userId` / `traits.userId` / `traits.id` / `context.traits.userId` / `context.traits.id` |  `cv_id` |
| 2 | `traits.email` / `context.traits.email` / `properties.email` | `cv_email` | 
| 3 | `context.externalId.woodpraId` / `anonymousId` | `cookie` |

<div class="infoBlock">
For a detailed list of the supported property mappings, refer to the <Link to="#standard-property-mappings">Standard property mappings</Link> section below.
</div>

> Any user-provided cookie must be mapped to either `email` or `cv_id` when creating a new user or using the cookie for the first time. Otherwise, Woopra accepts the data but will reflect it **only when** that cookie is mapped to any of these identifiers. For example, Woopra accepts the following event but does not reflect the data in the dashboard:

Reviewer 2: not cv_id ( as it is for woopra side ) it would be userId (Rudder property).

> Any user-provided cookie must be mapped to either `email` or `cv_id` when creating a new user or using the cookie for the first time. Otherwise, Woopra accepts the data but will reflect it **only when** that cookie is mapped to any of these identifiers. For example, Woopra accepts the following event but does not reflect the data in the dashboard:

Author: ```suggestion
Any user-provided cookie must be mapped to either `email` or `userId` when creating a new user or using the cookie for the first time. Otherwise, Woopra accepts the data but will reflect it **only when** that cookie is mapped to any of these identifiers. For example, Woopra accepts the following event but does not reflect the data in the dashboard:
```

> Any user-provided cookie must be mapped to either `email` or `userId` when creating a new user or using the cookie for the first time. Otherwise, Woopra accepts the data but will reflect it **only when** that cookie is mapped to any of these identifiers. For example, Woopra accepts the following event but does not reflect the data in the dashboard:

Reviewer 3: ```suggestion
Any user-provided cookie must be mapped to either `email` or `userId` when creating a new user or using the cookie for the first time. Otherwise, Woopra accepts the data but reflects it **only when** that cookie is mapped to any of these identifiers. For example, Woopra accepts the following event but does not reflect the data in the dashboard:
```

```javascript
rudderanalytics.identify({
  name: "Alex Keener",
  country: "USA",
} {
  externalId: [{
    id: "some_external_id_1",
    type: "woopraId",
  }, ],
});
```

> When you include a user identifier (`email`) in the subsequent events as shown below, Woopra reflects all the user information present in the current and previous event.

Reviewer 3: ```suggestion
When you include a user identifier (`email`) in the subsequent events as shown below, Woopra reflects all the user information present in the current and previous events.
```

```javascript
rudderanalytics.track("Order Completed", {
  orderId: "1234567",
  price: "567",
  currency: "USD",
});
```

### Supported mappings

The following table lists the supported **optional** property mappings between RudderStack and Woopra for the `track` events:

| RudderStack property | Woopra property | Data type |
| :-----| :------| :-----|
| `event` | `event` | String (in lower case) | 
| `originalTimestamp` | `timestamp` | Unix timestamp (in ms) | 
| `properties.{x}` | `ce_{x}` | - |

<div class="infoBlock">
If the event name is empty, RudderStack sends an empty string to Woopra as well, as a default fallback.
</div>

## Page

When you send a <Link to="/event-spec/standard-events/track/">`page`</Link> event, RudderStack converts it to a Woopra [`track`](https://docs.woopra.com/reference/track-ce#pageviews) event and sends all the page-related information and the associated properties.

### Supported mappings

The following table lists the supported **optional** property mappings between RudderStack and Woopra for the `page` events:

| RudderStack property | Woopra property | Data type |
| :-----| :------| :-----|
| `properties.{x}` | `ce_{x}` | - |

> | `name` | `event` | String (in lower case) |

Reviewer 2: ```suggestion
| `Viewed `category` `name` Page` | `event` | String (in lower case) |
```

> | `Viewed <category name> Page` | `event` | String (in lower case) |

Author: ```suggestion
| `Viewed <category> <name> Page` | `event` | String (in lower case) |
```

## Standard property mappings

The following table lists the supported property mappings between RudderStack and Woopra, **common to all the above events**:

| RudderStack property | Woopra property | Data type | Presence |
| :-----| :------| :-----| :------|
| `integration.woopra.propertyName` | `property` | Domain | Optional |
| `userId` / `traits.userId` / `traits.id` / `context.traits.userId` / `context.traits.id` | `cv_id` | String | Required, if `cv_email` or `cookie` is absent. |
| `traits.email` / `context.traits.email` / `properties.email` | `cv_email` | String | Required, if `cv_id` or `cookie` is absent. |
| `context.externalId.woodpraId` / `anonymousId` | `cookie` | String | Required, if `cv_email` or `cv_id` is absent. |
| `context.ip` | `cv_ip` | IP address | Optional |
| `traits.company` | `cv_company` | String | Optional |
| `traits.name` / `context.traits.name` | `cv_name` | String | Optional |

> | `traits.{x}` / `context.traits.{x}` | - | Optional, sent to Woopra as custom fields. |

Reviewer 3: ```suggestion
| `traits.{x}` / `context.traits.{x}` | - | Custom fields | Optional |
```

| RudderStack property | Woopra property | Data type |
| :-----| :------| :-----|
| `properties.{x}` | `ce_{x}` | - |

> | `Viewed <category name> Page` | `event` | String (in lower case) |

Reviewer 2: Can we change this to `Viewed <category> <name> Page` as category and name are two separate fields.

Author: Sure, we can.
